### PR TITLE
Tolerate errors in config file

### LINKED ISSUE #9577: unknown compiler options should not stop parsing of tsconfig.json
If a tsconfig.json has an unknown option, the whole tsconfig.json is thrown away. This is too drastic. instead, only unknown properties should be ignored, and other properties kept.

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #9577

Now uses the TypeScript parser to parse tsconfig.json to be more tolerating

### CODE REVIEW

File: src/server/editorServices.ts [hunk 1]
```diff
@@ -1341,20 +1348,19 @@ namespace ts.server {
...
                 Debug.assert(!!parsedCommandLine.fileNames);
 
-                if (parsedCommandLine.errors && (parsedCommandLine.errors.length > 0)) {
-                    return { succeeded: false, errors: parsedCommandLine.errors };
-                }
-                else if (parsedCommandLine.fileNames.length === 0) {
+                if (parsedCommandLine.fileNames.length === 0) {
                     const error = createCompilerDiagnostic(Diagnostics.The_config_file_0_found_doesn_t_contain_any_source_files, configFilename);
-                    return { succeeded: false, errors: [error] };
+                    return { succeeded: false, errors: concatenate(parsedCommandLine.errors, [error]) };
```
Comment (mhegazy) [inline src/server/editorServices.ts]: do you still need the succeeded property? it seems that now we are handling the two branches identically.

File: src/server/editorServices.ts [hunk 2]
```diff
@@ -1433,7 +1440,7 @@ namespace ts.server {
                     return watchers;
                 }, <Map<FileWatcher>>{});
 
-                return { success: true, project: project, errors };
+                return { success: true, project: project, errors: concatenate(errors, errorsFromConfigFile) };
```
Comment (mhegazy) [inline src/server/editorServices.ts]: same here, do you need success here?

File: src/server/editorServices.ts (hunk 1 above)
Comment (zhengbli) [inline src/server/editorServices.ts]: They are still different though. `succeeded` is true means we can proceed with building projects even if some minor errors exist, but `succeeded` is false means the errors are too serious that we are not going to build anything. Either invalid json and empty project will result in the later case.

File: src/server/editorServices.ts (hunk 2 above)
Comment (zhengbli) [inline src/server/editorServices.ts]: The same as above.

File: src/server/editorServices.ts (hunk 1 above)
Comment (mhegazy) [inline src/server/editorServices.ts]: even if it is invalid, why should not we build one? we found  tsconfig.json, and we know the open file. so we can create a project with just that..

**Comment by mhegazy** [comment]
👍

**Pull request merged**